src/reconcile/rec_visibility.c

```diff
@@ -327,6 +335,10 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
      * stable.
      */
     if (upd != NULL) {
+        /* Mark the prepare flag if the selected update is an uncommitted prepare. */
+        if (upd->prepare_state == WT_PREPARE_INPROGRESS || upd->prepare_state == WT_PREPARE_LOCKED)
+            upd_select->prepare = true;
```

Comment:
I feel upd_select->prepare may be redundant. Can we remove that?

Reply:
I feel it is not redundant. In the case of a prepared TOMBSTONE, the update returned is not a prepared update, so we will lose the information if we don't set the flag.

Reply:
That makes sense.

Reply:
Can you add a comment here to explain this?

Reply:
Added.